networkx/generators/stochastic.py

```diff
@@ -42,8 +42,5 @@ def stochastic_graph(G, copy=True, weight="weight"):
     # inside the loop may be costly in computation time.
     degree = dict(G.out_degree(weight=weight))
     for u, v, d in G.edges(data=True):
-        if degree[u] == 0:
```

Comment:
Are we 100% sure that this branch is un-hittable? What about cases involving self-loop edges? Isolated nodes? Multi edges?

It's important to really explore the input space when deleting conditionals like this, especially since the original motivation for the deletion is that the code isn't tested!

Reply:
This is indeed tricky, the only way this code could be hit if `G.edges` returns inverted edge for a directed graph (which itself would be a problem). If a node has self loop, it's out degree will be 1. For isolated node it won't ever show up in the G.edges.

This bit of code was added in https://github.com/networkx/networkx/pull/1001 but I'm not sure why this would ever happen in `G.edges`. This could happen if we were iterating over `G.nodes`!

This is a good catch, thanks @paulitapb!

Reply:
You're welcome @MridulS! I checked all those cases before deleting the conditional. Maybe I could have added these tests before deleting the conditional. I will have that in mind.

Reply:
FWIW I was being intentionally (probably annoyingly) vague in order to leave this as an exercise - there is indeed a valid case where this branch can be hit!

Reply:
oh perfect! I will try to find it!